src/workflow_tasks.rs

```diff
@@ -149,6 +155,8 @@ impl WorkflowTaskManager {
             });
         if let Some(act) = maybe_act.as_ref() {
             self.insert_outstanding_activation(&act);
+            // Evict from the workflow cache to ensure that run id is not going to be invalidated before activation is complete.
+            self.cache_manager.write().evict(&act.run_id);
```

Comment:
This should be touch, no? Explicitly evicting a run we just queued up an activation for seems wrong. Should have test for this as well

Reply:
As per our offline discussion, replaced evicts with touches and existing touch with insert.